.github/scripts/build.sh

```diff
@@ -0,0 +1,84 @@
+#!/bin/bash
+
+set -eux
+
+. .github/scripts/prereq.sh
+. .github/scripts/common.sh
+
+uname -a
+uname -p
+uname
+pwd
+env
+
+# ensure ghcup
+if ! command -v ghcup ; then
+	install_ghcup
+fi
+
+# ensure cabal-cache
+download_cabal_cache "$HOME/.local/bin/cabal-cache"
+
+
+# build
+ecabal update
+
+case "$(uname)" in
+    MSYS_*|MINGW*)
+		for ghc in $(cat bindist/ghcs-Msys) ; do
+			GHC_VERSION="${ghc%,*}"
+			args=( -O2 -w "ghc-$GHC_VERSION" --project-file cabal.project --disable-profiling --disable-tests --enable-executable-stripping ${ADD_CABAL_ARGS})
+			ghcup install ghc "${GHC_VERSION}"
+			ghcup set ghc "${GHC_VERSION}"
+			"ghc-${GHC_VERSION}" --info
+			"ghc" --info
+			# Shorten binary names
```

Comment:
Explanation why we shorten the binary names, maybe copied and pasted from other places, or we create a docs folder where this is explained in more detail in general. If the latter is desired, just copy-paste the comment/explanation for now, and create an issue.

Reply:
Because of MAX_PATH issues.